libnavui-status/src/main/res/layout/mapbox_status_view_layout.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    tools:parentTag="android.widget.FrameLayout">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/container"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:paddingTop="@dimen/mapbox_statusView_padding_v"
+        android:paddingBottom="@dimen/mapbox_statusView_padding_v"
+        android:paddingStart="@dimen/mapbox_statusView_padding_h"
+        android:paddingEnd="@dimen/mapbox_statusView_padding_h"
+        tools:background="@drawable/mapbox_bg_rounded">
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/messageText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:elevation="0dp"
+            android:ellipsize="end"
+            android:lines="1"
+            android:gravity="start|center_vertical"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            tools:text="Rerouting" />
+
+        <androidx.appcompat.widget.AppCompatImageView
+            android:id="@+id/iconImage"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_marginEnd="@dimen/mapbox_statusView_spacing"
+            android:visibility="visible"
+            app:layout_constraintStart_toEndOf="@+id/progressBar"
+            app:layout_constraintEnd_toStartOf="@id/messageText"
+            app:layout_constraintTop_toTopOf="@+id/messageText"
+            app:layout_constraintBottom_toBottomOf="@+id/messageText"
+            app:layout_constraintDimensionRatio="1:1"
+            tools:visibility="visible"
+            tools:srcCompat="@android:drawable/ic_secure" />
+
+        <ProgressBar
+            android:id="@+id/progressBar"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:indeterminate="true"
+            android:layout_marginEnd="@dimen/mapbox_statusView_spacing"
+            android:visibility="gone"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@+id/iconImage"
+            app:layout_constraintTop_toTopOf="@+id/messageText"
+            app:layout_constraintBottom_toBottomOf="@+id/messageText"
+            app:layout_constraintDimensionRatio="1:1"
+            android:layout_marginTop="2dp"
+            android:layout_marginBottom="2dp"
```

Comment:
![Screen Shot 2022-01-24 at 9 12 51 AM](https://user-images.githubusercontent.com/9770186/150831208-4a1837c9-0dbb-476e-abb9-f16a61c48a42.png)

Just wondering why do we need the `marginTop` and `marginBottom`? With these margin values the spinner is not aligned to the text view on the top.

Reply:
Just a personal preference. It looked a bit oversized to me so I decreased its size by padding it vertically.
It is aligned to a TextView center line.
Would you like me to change this?

Reply:
I think this is a developer preference. If it's related to the sizing, we can still align the view to the center of text view. At the moment, looks like only bottom is aligned.